test/firewall/firewall.go

```diff
@@ -0,0 +1,233 @@
+		defer func() {
+			err := utilsHelpers.DeletePod(debugPod)
+			if err != nil {
+				fmt.Printf("failed cleaning debug pod %s: %v", debugPod, err)
+			}
+		}()
+
+		command := []string{
+			"bash", "-c",
+			fmt.Sprintf("mkdir -p nft && echo '%s' > nft/firewall.nft", string(NFTtable)),
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save rule set to file on node %s: %w", node.Name, err)
+		}
+
+		installCmd := []string{"bash", "-c", "yum update -y && yum install -y nftables"}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, installCmd)
+		if err != nil {
+			return fmt.Errorf("failed to install nftables on  debugpod on node%s: %w", node.Name, err)
+		}
+
+		command = []string{
+			"bash", "-c", "/usr/sbin/nft -f nft/firewall.nft",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to apply rule set %q on node %s: %w", command, node.Name, err)
+		}
+
+		// Check the output of nft list ruleset
+		command = []string{"bash", "-c", "/usr/sbin/nft list ruleset"}
+		output, err := utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to list NFT ruleset on node %s: %w", node.Name, err)
+		}
+
+		if len(output) == 0 {
+			return fmt.Errorf("no output from 'nft list ruleset' on node %s: ", node.Name)
+		}
+		command = []string{
+			"bash", "-c",
+			"/usr/sbin/nft list ruleset > /etc/nftables.conf",
+		}
+
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to save NFT ruleset to file on node %s: %w", node.Name, err)
+		}
+
+		command = []string{"bash", "-c",
+			"systemctl enable nftables",
+		}
+		_, err = utilsHelpers.RunCommandOnPod(debugPod, command)
+		if err != nil {
+			return fmt.Errorf("failed to enable nftables %q on node %s: %w", command, node.Name, err)
+		}
+
+		artifactsDir, ok := os.LookupEnv("ARTIFACT_DIR")
+		if !ok {
+			fmt.Print("env var ARTIFACT_DIR is not set")
+		}
+
+		comArtifactsDir := filepath.Join(artifactsDir, "commatrix")
+		err = os.MkdirAll(comArtifactsDir, 0755)
+		if err != nil {
+			return fmt.Errorf("failed to create artifactsDir %w", err)
+		}
+
+		// save the nft ruleset on ArtifactDirPath.
+		err = os.WriteFile(filepath.Join(comArtifactsDir, fmt.Sprintf("nftables-%s.nft", node.Name)), output, 0644)
+		if err != nil {
+			return fmt.Errorf("failed to save nft file on for node artifacts %s %w", node.Name, err)
+		}
+	}
+	return nil
+}
+
+// isSNOCluster will check if OCP is a single node cluster.
+func isSNOCluster(oc *configv1client.ConfigV1Client) (bool, error) {
+	infrastructureType, err := oc.Infrastructures().Get(context.Background(), "cluster", metav1.GetOptions{})
+	if err != nil {
+		return false, err
+	}
+
+	return infrastructureType.Status.ControlPlaneTopology == configv1.SingleReplicaTopologyMode, nil
+}
+
+func infraType(oc *configv1client.ConfigV1Client) (bool, error) {
```

Comment:
The name of the func is `infraType` while it doesn't return the infra type, it's checking if it's BM and return bool